controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -222,29 +322,49 @@ func (r *StorageConsumerReconciler) own(resource metav1.Object) error {
 func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
 		func(context context.Context, obj client.Object) []reconcile.Request {
-			labels := obj.GetLabels()
-			if value, ok := labels[StorageConsumerNameLabel]; ok {
-				return []reconcile.Request{{
+			// Get the StorageConsumer objects
+			consumers := &ocsv1alpha1.StorageConsumerList{}
+			err := r.Client.List(context, consumers, &client.ListOptions{Namespace: obj.GetNamespace()})
+			if err != nil {
+				r.Log.Error(err, "Unable to list StorageConsumers")
+				return []reconcile.Request{}
+			}
+
+			// Return name and namespace of the StorageClusters object
+			request := []reconcile.Request{}
+			for _, sc := range consumers.Items {
+				request = append(request, reconcile.Request{
 					NamespacedName: types.NamespacedName{
-						Name:      value,
-						Namespace: obj.GetNamespace(),
+						Namespace: sc.Namespace,
+						Name:      sc.Name,
 					},
-				}}
+				})
 			}
```

Comment:
Don't create a copy of each consumer and use builtins where possible

```suggestion
	for i := range consumers.Items {
		request = append(
			request, 
			reconcile.Request{
				NamespacedName: client.ObjectKeyFromObject(consumer.items[i])
			})
		}
	}
	```